### DISCUSSION TITLE
Please comment/document the Anytype code

### DISCUSSION THREAD

**Discussion by Shampra** [anyproto/contributors#70] [Anytype apps]
Just... please comment your code, document function (with docstring or anything else), ...
it's probably neither awkward nor necessary between you, but by opening the code, volunteers to help should be able to understand it easily!
Especially if you want to create an active community around it.

**User case :** 
I loved spending my evening just testing out the functions to see what the damn "id" corresponds to... no hard feelings, but really, it would be useful and welcome!
💖

**Comment by ra3orblade** [comment]
Would love to but really do not have time. You can feel free to write me on Slack or telegram with some questions.

**Comment by Shampra** [comment]
@ra3orblade, I totally understand (and I'd rather the team worked on great functionality than on documentation).

And thanks for the suggestion.
But I'd like to avoid it, because it's also wasting time instead of saving it. With few contributors, it might not be a problem, but I'd like to see the community helping more and more with the code.
And then, a doc will be useful and everyone asking for the same things every time will be a waste of time.

_(well, now I'm behaving in an incomprehensible way, bug or not... if I don't find it I'll make a post)_

So the idea is:
- when adding or modifying a function, hop a little docstring in passing
- or if someone in the community feels motivated and confident enough to comment or document (in particular Stores functions, or object architecture doc - protobuf may answer that, I don't know enough about it)

The more external people want to participate, the more useful it will be ;-)

**Comment by joceqo** [comment]
Hi, I would love to make some PR to add docs, I really like the project and even though I don't have the skills to make a some PR like adding new block or such, I up for making docs PR.

**Reply by ra3orblade** [reply]
You can do it, but I thought to add it with AI agent after the next release, would take couple of hours.
